.github/actions/upload/src/filecoin.js

```diff
@@ -0,0 +1,262 @@
+import { promises as fs } from 'node:fs'
+import { RPC_URLS, TIME_CONSTANTS } from '@filoz/synapse-sdk'
+import { ethers } from 'ethers'
+import { createCarFromPath } from 'filecoin-pin/dist/add/unixfs-car.js'
+import { validatePaymentSetup } from 'filecoin-pin/dist/common/upload-flow.js'
+import {
+  checkAndSetAllowances,
+  computeTopUpForDuration,
+  depositUSDFC,
+  getPaymentStatus,
+} from 'filecoin-pin/dist/synapse/payments.js'
+import {
+  cleanupSynapseService,
+  createStorageContext,
+  initializeSynapse as initSynapse,
+} from 'filecoin-pin/dist/synapse/service.js'
+import { getDownloadURL, uploadToSynapse } from 'filecoin-pin/dist/synapse/upload.js'
+import { formatRunwayDuration } from 'filecoin-pin/dist/utils/time.js'
+import { CID } from 'multiformats/cid'
+import { ERROR_CODES, FilecoinPinError, getErrorMessage } from './errors.js'
+
+/**
+ * @typedef {import('./types.js').ParsedInputs} ParsedInputs
+ * @typedef {import('./types.js').BuildResult} BuildResult
+ * @typedef {import('./types.js').UploadResult} UploadResult
+ * @typedef {import('./types.js').PaymentStatus} PaymentStatus
+ * @typedef {import('./types.js').FilecoinPinPaymentStatus} FilecoinPinPaymentStatus
+ * @typedef {import('@filoz/synapse-sdk').Synapse} Synapse
+ */
+
+/**
+ * Calculate storage runway based on current payment status
+ * @param {FilecoinPinPaymentStatus} status - Payment status from getPaymentStatus
+ * @returns {string} Formatted runway duration or 'Unknown'
+ */
+export function calculateStorageRunway(status) {
```

Comment:
some of this functionality seems like it might be in the newer work around fund and auto-fund so doesn't need to be repeated?